motor/platform/detector/detector.go

```diff
@@ -89,20 +88,8 @@ func (d *Detector) Platform() (*platform.Platform, error) {
 		}, nil
 	case *google.Provider:
 		return pt.PlatformInfo()
-	case *azure.Provider:
-		return &platform.Platform{
-			Name:    "azure",
-			Title:   "Microsoft Azure",
-			Kind:    providers.Kind_KIND_API,
-			Runtime: providers.RUNTIME_AZ,
-		}, nil
-	case *ms365.Provider:
-		return &platform.Platform{
-			Name:    "microsoft365",
-			Title:   "Microsoft 365",
-			Kind:    providers.Kind_KIND_API,
-			Runtime: providers.RUNTIME_MICROSOFT_GRAPH,
-		}, nil
+	case *microsoft.Provider:
```

Comment:
I think we want to move forward and take the same approach for all providers. For os detection, we probably can migrate that into the os provider. This is a follow-up task, nothing to do as part of this PR.